src/test/java/teammates/storage/sqlapi/NotificationsDbTest.java

```diff
@@ -98,4 +99,49 @@ public void testCreateNotification_invalidNonNullParameters_emptyMessage() {
         assertThrows(InvalidParametersException.class, () -> notificationsDb.createNotification(invalidNotification));
         verify(session, never()).persist(invalidNotification);
     }
+
+    @Test
+    public void testGetNotification_success() throws EntityAlreadyExistsException, InvalidParametersException {
+        Notification notification = new Notification.NotificationBuilder()
+                .withNotificationId(UUID.randomUUID())
+                .withStartTime(Instant.parse("2011-01-01T00:00:00Z"))
+                .withEndTime(Instant.parse("2099-01-01T00:00:00Z"))
+                .withStyle(NotificationStyle.DANGER)
+                .withTargetUser(NotificationTargetUser.GENERAL)
+                .withTitle("A deprecation note")
+                .withMessage("<p>Deprecation happens in three minutes</p>")
+                .build();
+        notificationsDb.createNotification(notification);
+        verify(session, times(1)).persist(notification);
+
+        when(session.get(Notification.class, notification.getNotificationId())).thenReturn(notification);
+        Notification actualNotification = notificationsDb.getNotification(notification.getNotificationId());
+
+        assertEquals(notification, actualNotification);
+    }
+
+    @Test
+    public void testGetNotification_entityDoesNotExist() throws EntityAlreadyExistsException, InvalidParametersException {
+        Notification notification = new Notification.NotificationBuilder()
+                .withNotificationId(UUID.randomUUID())
+                .withStartTime(Instant.parse("2011-01-01T00:00:00Z"))
+                .withEndTime(Instant.parse("2099-01-01T00:00:00Z"))
+                .withStyle(NotificationStyle.DANGER)
+                .withTitle("A deprecation note")
+                .withTargetUser(NotificationTargetUser.GENERAL)
+                .withMessage("<p>Deprecation happens in three minutes</p>")
+                .build();
+        notificationsDb.createNotification(notification);
+        verify(session, times(1)).persist(notification);
+
+        UUID nonExistentId = UUID.randomUUID();
+        while (nonExistentId.equals(notification.getNotificationId())) {
```

Comment:
Might be better to "hardcode" the UUID here.

Reply:
I think it is better to create a private method in this class for generating different UUID like what you suggested.